controllers/pod_models.go

```diff
@@ -790,3 +755,139 @@ func GetHeadlessService(cluster *fdbtypes.FoundationDBCluster) (*corev1.Service,
 
 	return service, nil
 }
+
+// ensureContainerPresent looks for a container by name from a list, and adds
+// an empty container with that name if none is present.
+func ensureContainerPresent(containers []corev1.Container, name string, insertIndex int) ([]corev1.Container, int) {
+	for index, container := range containers {
+		if container.Name == name {
+			return containers, index
+		}
+	}
+
+	if insertIndex < 0 || insertIndex >= len(containers) {
+		containers = append(containers, corev1.Container{Name: name})
+		return containers, len(containers) - 1
+	} else {
+		containerCount := 1 + len(containers)
```

Comment:
This could be simplified with https://github.com/golang/go/wiki/SliceTricks#insert e.g.

```go
containers = append(containers, corev1.Container{})
copy(containers[insertIndex+1:], containers[insertIndex:])
containers[insertIndex] = corev1.Container{Name: name}
```
this has the benefit that we don't need to allocate a new slice.